spec/support/vcr_helper.rb

```diff
@@ -0,0 +1,10 @@
+module VCRHelper
+  def self.spinlock_until(&_block)
+    try_count = 0
+    loop do
+      try_count += 1
+      break if try_count >= 30 || yield
+      sleep 1
+    end
+  end
+end
```

Comment:
Usually for something like this I reach for ruby's `redo` or `retry`.  In this case though I might write this like 

```ruby
module VCRHelper
  def self.spinlock_until(retries: 30)
    retries.times do
      return if yield
      sleep(1)
    end
  end
end
```